app/models/run.rb

```diff
@@ -8,4 +8,41 @@ class Run < ApplicationRecord
 
   belongs_to :person
   belongs_to :race
+
+  validates :person_id, uniqueness: { 
+    scope: :race_id,
+    message: 'already registered for this race'
+  }
+  validate :person_availability
+  validate :person_eligibility
+  validate :race_ongoing_registration
+  validate :race_overlaps_registered
+
+  def person_availability
+    person = Person.find(person_id)
+    return unless person.ongoing_practice?
```

Comment:
for readability and mas intuitive, mas preferred namin na `if !` instead of `unless`. tho good thing din naman alam mo gamitin. mas makikita mo lang sya sa mga prod code namin

```suggestion
    return if !person.ongoing_practice?
```

Reply:
Resolved in this [branch/commit](https://github.com/scrambledeggs/iole-trails/pull/7/commits/0dccf19cd079450fd6e4ebba5a2ddf0eefcca7ee)